### PR TITLE
Add io.github.lainsce.Quilter

### PR DESCRIPTION
**Pull request by lainsce**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: io.github.lainsce.Quilter.json [hunk 1]
```diff
@@ -0,0 +1,106 @@
+{
+    "app-id" : "io.github.lainsce.Quilter",
+    "runtime" : "org.gnome.Sdk",
+    "runtime-version" : "3.38",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "io.github.lainsce.Quilter",
+    "rename-icon" : "io.github.lainsce.Quilter",
+    "finish-args" : [
+        "--filesystem=xdg-data/io.github.lainsce.Quilter/",
```
Comment (barthalion) [inline io.github.lainsce.Quilter.json:9]: Your application has write here by default, just needs to follow `$XDG_DATA_HOME`.

**Comment by barthalion** [comment]
bot, build io.github.lainsce.Quilter

**Comment by barthalion** [comment]
One thing to check is to make sure it works fine with Adwaita (looks fine to me at first glance) – otherwise you should enforce Elementary theme.

File: io.github.lainsce.Quilter.json [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    "modules" : [
+        {
+            "name": "gtksourceview",
+            "buildsystem": "meson",
+            "sources": [{
+                "type": "archive",
+                "url": "https://download.gnome.org/sources/gtksourceview/4.7/gtksourceview-4.7.90.tar.xz",
+                "sha256": "983bdcb88f98285b3b997c9335057c8fbc45aad0a7c13cb65eb365567e26baaf"
+            }]
+        },
```
Comment (bilelmoussaoui) [inline io.github.lainsce.Quilter.json:42]: gtksourceview4 is already on the sdk

File: io.github.lainsce.Quilter.json [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        {
+            "name" : "libhandy",
+            "buildsystem" : "meson",
+            "config-opts" : [
+                "-Dexamples=false",
+                "-Dprofiling=false",
+                "-Dglade_catalog=disabled",
+                "-Dintrospection=enabled",
```
Comment (bilelmoussaoui) [inline io.github.lainsce.Quilter.json:82]: you don't need introspection for vala.

**Comment by lainsce** [comment]
> One thing to check is to make sure it works fine with Adwaita (looks fine to me at first glance) – otherwise you should enforce Elementary theme.

It should be Adwaita, yes.

File: io.github.lainsce.Quilter.json (hunk 2 above)
Comment (lainsce) [reply io.github.lainsce.Quilter.json:42]: Oh, I see. Going to remove these lines then.

File: io.github.lainsce.Quilter.json (hunk 3 above)
Comment (lainsce) [reply io.github.lainsce.Quilter.json:82]: Oops, this was me porting the rust flatpak to vala.

**Comment by nedrichards** [comment]
bot, build io.github.lainsce.Quilter

**Comment by lainsce** [comment]
Had to do some small styling fixes. But nothing major.

**Comment by barthalion** [comment]
There's still one comment left about the usage of `--filesystem=xdg-data/io.github.lainsce.Quilter` – which shouldn't be needed in practice.

**Comment by nedrichards** [comment]
bot, build io.github.lainsce.Quilter

File: io.github.lainsce.Quilter.json (hunk 1 above)
Comment (lainsce) [reply io.github.lainsce.Quilter.json:9]: so just `--filesystem=xdg-data` instead?

File: io.github.lainsce.Quilter.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
     "command" : "io.github.lainsce.Quilter",
     "rename-icon" : "io.github.lainsce.Quilter",
     "finish-args" : [
-        "--filesystem=xdg-data/io.github.lainsce.Quilter/",
+        "--filesystem=xdg-data",
```
Comment (bilelmoussaoui) [inline io.github.lainsce.Quilter.json:9]:
```suggestion
```

**Comment by nedrichards** [comment]
bot, build io.github.lainsce.Quilter

**Comment by lainsce** [comment]
Sorry for making you guys test-build so much, but this last commit is the last one I'll do. :bow:

**Pull request closed without merging**